src/modules/ekf2/EKF/gps_control.cpp

```diff
@@ -117,25 +117,27 @@ void Ekf::controlGpsFusion()
 
 						/* A reset is not performed when getting GPS back after a significant period of no data
 						 * because the timeout could have been caused by bad GPS.
-						 * The total number of resets allowed per boot cycle is limited.
 						 */
 						if (isYawFailure()
 						    && _control_status.flags.in_air
 						    && !was_gps_signal_lost
-						    && _ekfgsf_yaw_reset_count < _params.EKFGSF_reset_count_limit
-						    && isTimedOut(_ekfgsf_yaw_reset_time, 5'000'000)) {
+						    && isTimedOut(_bad_gnss_fusion_yaw_reset_time, _params.reset_interval_min)) {
 							// The minimum time interval between resets to the EKF-GSF estimate is limited to allow the EKF-GSF time
 							// to improve its estimate if the previous reset was not successful.
 							if (resetYawToEKFGSF()) {
 								ECL_WARN("GPS emergency yaw reset");
+								_bad_gnss_fusion_yaw_reset_time = _imu_sample_delayed.time_us;
 							}
 						}
 
-						ECL_WARN("GPS fusion timeout, resetting velocity and position");
-						_information_events.flags.reset_vel_to_gps = true;
-						_information_events.flags.reset_pos_to_gps = true;
-						resetVelocityTo(gps_sample.vel, vel_obs_var);
-						resetHorizontalPositionTo(gps_sample.pos, pos_obs_var);
+						if (isTimedOut(_bad_gnss_fusion_gnss_reset_time, _params.reset_interval_min)) {
+							ECL_WARN("GPS fusion timeout, resetting velocity and position");
+							_information_events.flags.reset_vel_to_gps = true;
+							_information_events.flags.reset_pos_to_gps = true;
+							resetVelocityTo(gps_sample.vel, vel_obs_var);
+							resetHorizontalPositionTo(gps_sample.pos, pos_obs_var);
+							_bad_gnss_fusion_gnss_reset_time = _imu_sample_delayed.time_us;
```

Comment:
If we use GNSS vel/pos time_last_fuse for the timeouts and set time_last_fuse on vel/pos reset then we don't need to carry the extra `_bad_gnss_fusion_gnss_reset_time`.

```suggestion
							_aid_src_gnss_pos.time_last_fuse = _imu_sample_delayed.time_us;
```

Reply:
I was thinking of the case where the GNSS fusion starts in air (_aid_src_gnss_pos.time_last_fuse is set to "now") but with a bad yaw that makes the fusion fail. In that case, it would then require 7s (interval) instead of 4s